src/test/java/com/example/SS2_Backend/service/MTMStableMatchingSolverTest.java

```diff
@@ -0,0 +1,66 @@
+package com.example.SS2_Backend.service;
+
+import com.example.SS2_Backend.dto.mapper.StableMatchingProblemMapper;
+import com.example.SS2_Backend.dto.request.NewStableMatchingProblemDTO;
+import com.example.SS2_Backend.ss.smt.evaluator.impl.TwoSetFitnessEvaluator;
+import com.example.SS2_Backend.util.MatchingProblemType;
+import com.example.SS2_Backend.util.SampleDataGenerator;
+import jakarta.validation.ConstraintViolation;
+import jakarta.validation.Validation;
+import jakarta.validation.Validator;
+import jakarta.validation.ValidatorFactory;
+import org.junit.Test;
+import org.junit.jupiter.api.BeforeEach;
+import org.moeaframework.core.NondominatedPopulation;
+
+import java.util.Objects;
+import java.util.Set;
+
+import static org.junit.Assert.assertFalse;
+
+public class MTMStableMatchingSolverTest {
+    NewStableMatchingProblemDTO newStableMatchingProblemDTO;
+    SampleDataGenerator sampleData;
+    int numberOfIndividuals1;
+    int numberOfIndividuals2;
+    int numberOfProperties;
+    private Validator validator;
+
+    @BeforeEach
+    public void setUp() {
+        numberOfIndividuals1 = 20;
+        numberOfIndividuals2 = 200;
+        sampleData = new SampleDataGenerator(
+                MatchingProblemType.MTM,
+                numberOfIndividuals1, numberOfIndividuals2,
+                numberOfProperties
+        );
+        newStableMatchingProblemDTO = sampleData.generateDto();
+        ValidatorFactory factory = Validation.buildDefaultValidatorFactory();
+        validator = factory.getValidator();
+
+    }
+
+    @Test
+    public void testEvaluateFunctions() {
```

Comment:
Không có case assert tính sai nhé, test funtional không phải coverage

Chuyển thành: Set cho một Evaluate function đúng
assert kết quả trả ra đúng như tính toán

Reply:
Em để luôn là một String array toàn "default" nha